Activity Feed MVP

ananda1066 opened the pull request:
Activity Feed MVP completed. Created new Activity class and ActivityStore class. Now, Conversation, Chat, Message extend Activity and ActivityStore creates list of activities. Looking to add user profiles and friends features in the future possibly. Demo at: alisha-imhere.appspot.com

srjoglekar246 commented on src/main/java/codeu/controller/LoginServlet.java:
```diff
@@ -77,6 +90,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
+    // set type of activity to 1 (user logged in) and add to activity list
+    user.setType(1);
+    activityStore.addActivity(user);
+
```
Reduce number of newlines to 1

srjoglekar246 commented on src/main/java/codeu/model/data/Activity.java:
```diff
@@ -0,0 +1,50 @@
+  public String formattedTime(){
+    Date date = Date.from(this.getCreationTime());
+    SimpleDateFormat formatter = new SimpleDateFormat("MM-dd-yyyy HH:mm:ss");
+    return formatter.format(date) + " PST";
```
`return formatter.format(Date.from(this.getCreationTime())) + " PST";` since date is only used once.

srjoglekar246 commented on src/main/java/codeu/model/data/Activity.java:
```diff
@@ -0,0 +1,50 @@
+  public abstract String getTitle();
+
+  /**
+    Types distinguish activity type:
```
Better to have an enum for this, will make the code more readable.

srjoglekar246 commented on src/main/java/codeu/model/data/User.java:
```diff
@@ -18,11 +18,12 @@
 /** Class representing a registered user. */
-public class User {
-  private final UUID id;
+public class User extends Activity{
```
A `User` extending `Activity` doesn't make sense. You seem to be using `User` for 'login' as an activity. If yes, it might make more sense to have that as a separate class (or do it in a future PR, skip login tracking for now).

ananda1066 commented:
Changed design so Activity is its own class and just associates with objects using their UUIDs. Probably a better design since Conversations and Messages aren't really Activities either; they're just doing the Activities. Also, updated PersistentDataStore to save activities so each user can see all previous activity even when they were logged out. And added all tests, for object class, servlet, store, and updated tests for PersistentDataStore.

srjoglekar246 commented on src/main/java/codeu/controller/ConversationServlet.java:
```diff
@@ -131,9 +132,8 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
-    conversation.setType(2);
-    activityStore.addActivity(conversation);
+    // create activity with type NEW_CONVERSATION and add to activity list
+    activityStore.addActivity(new Activity(Activity.Type.NEW_CONVERSATION, conversation.getId(), conversation.getCreationTime(), UUID.randomUUID()));
```
As a good convention, keep the column length (no. of characters on each line) restricted to 100: https://google.github.io/styleguide/javaguide.html#s4.4-column-limit

Here & everywhere. This might help: https://github.com/google/google-java-format

srjoglekar246 commented on src/main/java/codeu/model/data/Activity.java:
```diff
@@ -17,34 +17,74 @@
+  // type of Activity
+  public enum Type{
+    REGISTERED, LOGGED_IN, NEW_CONVERSATION, NEW_MESSAGE
+  };
```
newline between enum and variables

srjoglekar246 commented on src/test/java/codeu/model/store/basic/ActivityStoreTest.java:
```diff
@@ -0,0 +1,52 @@
+
+  @Test
+  public void testAddActivity() {
+    Activity inputActivity =
```
Club lines 34 & 37 together.
Then do `Activity resultActivity = activityStore.getAllActivities().get(1);`

This way, the test will have 3 blocks: 1) All inits 2) Behavior being tested 3) Verifications on the behavior

srjoglekar246 commented on src/test/java/codeu/model/store/basic/ActivityStoreTest.java:
```diff
@@ -0,0 +1,52 @@
+    Mockito.verify(mockPersistentStorageAgent).writeThrough(inputActivity);
+  }
+
+  private void assertEquals(Activity expectedActivity, Activity actualActivity) {
```
assertEquals is a little too generic, in the test it might not be clear if you are using a custom function or something off a library. Rename to something like `assertEquivalentActivities`

srjoglekar246 commented on src/test/java/codeu/model/store/persistence/PersistentStorageAgentTest.java:
```diff
@@ -72,4 +79,12 @@ public void testWriteThroughMessage() {
+  public void testWriteThroughActivity() {
+    Activity activity =
+        new Activity(Activity.Type.LOGGED_IN, UUID.randomUUID(), Instant.now(), UUID.randomUUID());
+    persistentStorageAgent.writeThrough(activity);
```
insert empty line here.

srjoglekar246 commented:
Be careful with the conflicts :-)
